build_projects/dotnet-host-build/MsiTargets.cs

```diff
@@ -162,6 +162,7 @@ public static BuildTargetResult GenerateDotnetHostFxrMsi(BuildTargetContext c)
             var inputDir = c.BuildContext.Get<string>("HostFxrPublishRoot");
             var wixObjRoot = Path.Combine(Dirs.Output, "obj", "wix", "hostfxr");
             var hostFxrBrandName = $"'{Monikers.GetHostFxrBrandName(c)}'";
+            var upgradeCode = Utils.GenerateGuidFromName(HostFxrMsi).ToString().ToUpper();
```

Comment:
MSI name has nomenclature as dotnet-hostfxr-win-{arch}.{majorversion}.{minorversion}.{patchversion}-{releasesuffix}-{buildnumber} . We want to support SxS for build to build install within a release and across releases. So, it will be unique always. What would be case when we wouldn't want to change the upgradecode ?